aodntools/timeseries_products/velocity_aggregated_timeseries.py

```diff
@@ -0,0 +1,316 @@
+            for dim in VoIdimensions:
+                if dim not in allowed_dimensions:
+                    error_list.append('not allowed dimension: ' + dim)
+
+    if 'NOMINAL_DEPTH' not in file_variables and 'instrument_nominal_depth' not in attributes:
+        error_list.append('no NOMINAL_DEPTH')
+
+    return error_list
+
+
+def get_nvalues(nc):
+    """
+    Get the number of cells above the sensor
+    :param nc: xarray dataset
+    :return: number of cells, number of bins
+    """
+    if 'HEIGHT_ABOVE_SENSOR' in nc.dims:
+        nbins = nc.dims['HEIGHT_ABOVE_SENSOR']
+        nvalues = nc.dims['TIME'] * nbins
+    else:
+        nbins = 1
+        nvalues = nc.dims['TIME']
+    return nvalues, nbins
+
+
+def get_varvalues(nc, varname):
+    """
+    Return a 1D array of 2D values
+    :param nc: dataset
+    :param varname: Variable of interest
+    :return: variable values flattened
+    """
+    return nc[varname].values.flatten()
+
+
+def get_instrumentID(nc):
+    """
+    Create instrument id based on deployment metadata
+    :param nc: xarray dataset
+    :return: instrumentID as string
+    """
+    return '; '.join([nc.deployment_code, nc.instrument, nc.instrument_serial_number])
+
+
+def in_water(nc):
+    """
+    cut data to in-water only timestamps, dropping the out-of-water records.
+    :param nc: xarray dataset
+    :return: xarray dataset
+    """
+    time_deployment_start = np.datetime64(nc.attrs['time_deployment_start'][:-1])
+    time_deployment_end = np.datetime64(nc.attrs['time_deployment_end'][:-1])
+    TIME = nc['TIME'][:]
+    return nc.where((TIME >= time_deployment_start) & (TIME <= time_deployment_end), drop=True)
+
+
+
+
+## MAIN FUNCTION
+def aggregate_velocity(files_to_agg, site_code, base_path):
+    """
+    Aggregate U, V and W CUR variables from all deployments at one site.
+    the vertical cells are flattened and related to its depth
+    additional metadata variables are stored to track the origin of the data
+    :param files_to_agg: list of files to aggregate
+    :param site_code: site code
+    :param base_path: base path to store the resulting file
+    :return: name of the resulting file, list of rejected files
+    """
+
+    varlist = ['UCUR', 'VCUR', 'WCUR', 'DEPTH']
+    time_units="days since 1950-01-01 00:00:00 UTC"
+    time_calendar="gregorian"
+    epoch = np.datetime64("1950-01-01T00:00:00")
+    one_day = np.timedelta64(1, 'D')
+
+    varlen_list = []
+    bad_files = []
+    varlen_file = []
+    rejected_files = []
+
+    # default name for temporary file. It will be renamed at the end
+    outfile = 'Velocity_agg_tmp.nc'
+
+    ## sort the file list in chronological order
+    files_to_agg = sort_files(files_to_agg)
+
+    ## check files and get total number of flattened obs
+    for file in files_to_agg:
+        with xr.open_dataset(file) as nc:
+            ## clip to in water data only
+            nc = in_water(nc)
```

Comment:
in_water_only = in_water(nc)

Reply:
or alike

Reply:
dont hide the state mutation. In another words, do not reuse the name if the state changed - There is no copies here only views